nlp/2021_11_25_gpt3mix/nlp_gpt3mix.ipynb

```diff
@@ -0,0 +1,1827 @@
+        "                           data = json.dumps({\"prompt\": prompt, \n",
+        "                                              \"max_tokens\": 30,\n",
+        "                                              \"temperature\": 0.9,\n",
+        "                                              \"top_p\": 0.95}))\n",
+        "\n",
+        "  # get response and extract the generated text and label\n",
+        "  # the generated output will be in the form \"<text> (Sentiment: <label>)\"\n",
+        "  data = response.json()['choices'][0]['text'].split('\\n')[0].split('(Sentiment:')\n",
+        "\n",
+        "  if len(data) < 2:\n",
+        "    # the format of the response is invalid\n",
+        "    continue\n",
+        "\n",
+        "  text = data[0]\n",
+        "  label = data[1].split(')')[0].strip()\n",
+        "\n",
+        "  if label not in ['joy', 'anger', 'surprise']:\n",
+        "    # the format of the response is invalid\n",
+        "    continue\n",
+        "\n",
+        "  new_texts.append(text)\n",
+        "  new_labels.append(label2idx[label])\n",
+        "  iter += 1\n",
+        "\n",
+        "# define the synthetic dataset and save it to disk so as to prevent sending \n",
+        "# many api requests\n",
+        "synthetic_ds = Dataset.from_dict({'text': new_texts, 'label': new_labels})\n",
+        "synthetic_ds.save_to_disk('./drive/MyDrive/text_augmentation/gpt-3/10')"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "rPlF7dXKor6v"
+      },
+      "source": [
+        "# load the synthetic datasets with 10, 50, 100 and 200 samples\n",
+        "# run this if you do not want to generate samples on your own.\n",
+        "\n",
+        "if not os.path.isdir('./data'):\n",
+        "  !git clone https://github.com/ml6team/quick-tips.git\n",
+        "  !cd quick-tips && git checkout feature/nlp_gpt3mix\n",
+        "  !mv quick-tips/nlp/2021_11_25_gpt3mix/data ./data\n",
+        "  !rm -rf quick-tips\n",
+        "\n",
+        "synthetic_gpt3_10_ds = load_from_disk('./data/gpt-3/10')\n",
+        "synthetic_gpt3_50_ds = load_from_disk('./data/gpt-3/50')\n",
+        "synthetic_gpt3_100_ds = load_from_disk('./data/gpt-3/100')\n",
+        "synthetic_gpt3_200_ds = load_from_disk('./data/gpt-3/200')"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4b4GsUwfINep"
+      },
+      "source": [
+        "Now let's print some synthetic samples to examine their quality!"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "6irrOSPrKVU9",
+        "outputId": "026598d0-5521-42a2-bde4-d1e73d2703d1"
+      },
+      "source": [
+        "print(\"Dataset of 10 synthetic samples:\")\n",
```

Comment:
You could make a function to print random samples from different datasets. I'd imagine some people will want to just print samples to get a sense check of how the generated texts look like.